ollama/_client.py

```diff
@@ -1075,9 +1079,15 @@ def _copy_messages(messages: Optional[Sequence[Union[Mapping[str, Any], Message]
     )
 
 
-def _copy_tools(tools: Optional[Sequence[Union[Mapping[str, Any], Tool]]]) -> Iterator[Tool]:
-  for tool in tools or []:
-    yield Tool.model_validate(tool)
+def _copy_tools(tools: Optional[Sequence[Union[Mapping[str, Any], Tool, Callable]]] = None) -> Iterator[Tool]:
+  if not tools:
+    return []
```

Comment:
Shouldn't that be the case? It would mean that no tools get assigned on the request shown below (which is what is expected)

```py
        model=model,
        messages=[message for message in _copy_messages(messages)],
        tools=[tool for tool in _copy_tools(tools)],
        stream=stream,
 ```

https://github.com/ollama/ollama-python/pull/321/files/16c868a9375af412d5e4be21ef3060afa256faf6#diff-0f246a9c084dd2ef9d4a58c02fb818ac4a114c34877e558cac88ab351daae9eeR790-L788

Reply:
Sorry missed a word. Meant to say "it shouldn't be _necessary_"